tests/integration/targets/ndo_schema_template_bd_dhcp_policy/tasks/main.yml

```diff
@@ -0,0 +1,409 @@
+          - nm_update_ndo_schema_template_bd_dhcp_policy_different_template.current.ref is defined
+          - nm_update_ndo_schema_template_bd_dhcp_policy_different_template.current.dhcpOptionLabel.name == "ansible_test_option_policy_2"
+          - nm_update_ndo_schema_template_bd_dhcp_policy_different_template.current.dhcpOptionLabel.ref != nm_update_ndo_schema_template_bd_dhcp_policy_different_template.previous.dhcpOptionLabel.ref
+
+    # QUERY
+
+    - name: Create another template bd dhcp relay policy
+      cisco.mso.ndo_schema_template_bd_dhcp_policy: &create_ndo_schema_template_bd_dhcp_policy_2
+        <<: *create_ndo_schema_template_bd_dhcp_policy
+        dhcp_relay_policy: ansible_test_relay_policy_2
+        dhcp_relay_policy_template: ansible_tenant_template_2
+
+    - name: Query a template bd dhcp relay policy with template_name
+      cisco.mso.ndo_schema_template_bd_dhcp_policy:
+        <<: *create_ndo_schema_template_bd_dhcp_policy_2
+        state: query
+      register: query_one
+
+    - name: Query all template bd dhcp relay policy in the template
+      cisco.mso.ndo_schema_template_bd_dhcp_policy:
+        <<: *mso_info
+        schema: ansible_test
+        template: ansible_test_dhcp_policy
+        bd: ansible_test_bd_dhcp_policy
+        state: query
+      register: query_all
+    
+    - name: Assert template bd dhcp relay policy was queried
+      assert:
+        that:
+          - query_one is not changed
+          - query_one.current.name == "ansible_test_relay_policy_2"
+          - query_one.current.ref is defined
+          - query_one.current.dhcpOptionLabel is not defined
+          - query_all is not changed
+          - query_all.current | length == 2
+
+    # ERRORS
+
+    - name: Error dhcp_relay_policy without template
+      cisco.mso.ndo_schema_template_bd_dhcp_policy:
+        <<: *create_ndo_schema_template_bd_dhcp_policy
+        dhcp_relay_policy: '{{ fakevar | default(omit)}}'
+        dhcp_relay_policy_template: '{{ fakevar | default(omit)}}'
+      ignore_errors: true
+      register: err_relay_policy_without_template
+
+    - name: Error query a template bd dhcp relay policy that does not exist
+      cisco.mso.ndo_schema_template_bd_dhcp_policy:
+        <<: *create_ndo_schema_template_bd_dhcp_policy_2
+        dhcp_relay_policy: fake_relay_policy
+        state: query
+      ignore_errors: true
+      register: err_relay_policy_not_exist
+    
+    - name: Assert errors
+      assert:
+        that:
+          - err_relay_policy_without_template is failed
+          - err_relay_policy_without_template.msg == "state is present but all of the following are missing{{":"}} dhcp_relay_policy"
+          - err_relay_policy_not_exist is failed
+          - err_relay_policy_not_exist.msg == "Provided DHCP Relay Policy with '[KVPair(key='name', value='fake_relay_policy')]' not matching existing object(s){{":"}} ansible_test_relay_policy_2"
```

Comment:
Can we use `'[KVPair(key='name', value='fake_relay_policy')]'` in the error message?

Reply:
Is there a reason why we cannot?